rskj-core/src/main/java/co/rsk/peg/LockWhitelist.java

```diff
@@ -56,24 +67,23 @@ public Integer getSize() {
 
     public List<Address> getAddresses() {
         // Return a copy so that this can't be modified from the outside
-        return new ArrayList<>(whitelistedAddresses);
+        return Collections.unmodifiableList(new ArrayList<>(whitelistedAddresses.keySet()));
```

Comment:
You might want to modify the copy you are returning, that's up to the user. The only thing we want to ensure is that the internal instance is not modified.